src/System.Linq.Expressions/tests/New/NewTests.cs

```diff
@@ -466,16 +466,46 @@ public static IEnumerable<object[]> Type_InvalidType_TestData()
             yield return new object[] { typeof(int).MakeByRefType() };
             yield return new object[] { typeof(StaticCtor) };
             yield return new object[] { typeof(ClassWithNoDefaultCtor) };
+            yield return new object[] { typeof(int).MakePointerType() };
```

Comment:
@hughbe in adding types to this data-source, I added this and removed your InlineData attribute because while we can't do `typeof(int*)` in an iterator we can call `MakePointerType` which has the same result, and this seemed to be your intention. Was I right in deducing that?